core/templates/dev/head/domain/collection/CollectionObjectFactorySpec.js

```diff
@@ -324,6 +337,117 @@ describe('Collection object factory', function() {
     }
   );
 
+  it('should contain a collection skill defined in the backend object',
+    function() {
+      var collectionSkillBackendObject = {
+        name: 'skill01',
+        question_ids: []
+      };
+      var collection = CollectionObjectFactory.create({
+        id: 'collection_id',
+        nodes: [],
+        skills: {
+          skill0: collectionSkillBackendObject
+        },
+        next_skill_id: 1
+      });
+      expect(collection.containsCollectionSkill('skill0')).toBe(true);
+      expect(collection.containsCollectionSkill('fake skill')).toBe(false);
+      expect(collection.getCollectionSkills()).toEqual({
+        skill0: CollectionSkillObjectFactory.create(
+          'skill0', collectionSkillBackendObject)
+      });
+    }
+  );
+
+  it('should be able to add an existing skill object to the collection',
+    function() {
+      var collectionSkillBackendObject = {
+        name: 'skill01',
+        question_ids: []
+      };
+      _sampleCollection.setNextSkillId(1);
+      expect(_sampleCollection.addCollectionSkill(
+        CollectionSkillObjectFactory.create(
+          'skill0', collectionSkillBackendObject))).toBe(true);
+      expect(_sampleCollection.containsCollectionSkill('skill0')).toBe(true);
+    }
+  );
+
+  it('should be able to add new skills by skill name', function() {
+    expect(_addCollectionSkill('new skill')).toBe(true);
+    var newSkillId = _sampleCollection.getSkillIdFromName('new skill');
+    expect(newSkillId).toEqual('skill0');
+    expect(_sampleCollection.containsCollectionSkill(newSkillId)).toBe(true);
+
+    expect(_addCollectionSkill('new skill 2')).toBe(true);
+    var newSkillId2 = _sampleCollection.getSkillIdFromName('new skill 2');
+    expect(newSkillId2).toEqual('skill1');
+    expect(_sampleCollection.containsCollectionSkill(newSkillId2)).toBe(true);
```

Comment:
If ``containsCollectionSkill`` always returns true, will this test still pass? Should it still pass?